### ISSUE TITLE
panic: interface conversion: interface {} is nil, not map[string]interface {} in google_storage_bucket

### ISSUE THREAD

**Issue by olehmart** [hashicorp/terraform-provider-google#11404]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version
Terraform v1.1.7

- provider google v4.15.0

- provider google-beta v4.15.0

### Affected Resource(s)

* google_storage_bucket

### Terraform Configuration Files

```tf
resource "google_storage_bucket" "bucket" {
  project       = var.project_id
  name          = "bucket-name"
  location      = "US"
  force_destroy = true
  storage_class = "STANDARD"

  versioning {
    enabled = false
  }
  labels = {
    label1: "value1"
    label2: "value2"
  }
  logging {
    log_bucket = ""
    log_object_prefix = ""
  }
  encryption {
    default_kms_key_name = ""
  }
}
```

### Debug Output
[Debug output](https://gist.githubusercontent.com/olehmart/6a091d255bf48f0a083c07221669badc/raw/974f03037add35757b761e0e29acfc5f4db8fe17/debug_output.log)

### Panic Output
[Crash log](https://gist.githubusercontent.com/olehmart/272a50c8ddbb1a4b91cd29d77ad55f7d/raw/d844068f75d3ec3134bf805698684c67f2d20d5d/crash.log)

### Expected Behavior
When I run apply for this resource, it should create it. When I run apply for this resource one more time, it should keep it unchanged or update if there is some change.

### Actual Behavior

I run apply for the first time. Terraform creates it successfully. When I run terraform one more time it's failing.

### Steps to Reproduce

Run terraform apply. Then, run apply one more time.

b/275621288

**Comment by olehmart** [comment]
as a workaround I used the following to make it working properly:
```tf
dynamic "logging" {
    for_each = range(try(each.value.logging, null) != null ? 1 : 0)
    content {
      log_bucket        = each.value.logging_bucket
      log_object_prefix = each.value.logging_object_prefix
    }
  }
  dynamic "encryption" {
    for_each = range(try(each.value.encryption, null) != null ? 1 : 0)
    content {
      default_kms_key_name = each.value.default_kms_key_name
    }
  }
```
If logging or encryption configuration is provided, terraform will dynamically add appropriate parameters for `google_storage_bucket` module

**Comment by thomasmaclean** [comment]
My team and I are taking over for any issues related to Cloud Storage. We believe this issue was already fixed earlier as per c522bcf. Please reopen if this was closed incorrectly.

**Comment by rileykarson** [comment]
Duplicates https://github.com/hashicorp/terraform-provider-google/issues/13038 / closed as indicated
